server/api/groups.js

```diff
@@ -0,0 +1,250 @@
+// @flow
+import Router from 'koa-router';
+import auth from '../middlewares/authentication';
+import pagination from './middlewares/pagination';
+import { Op } from '../sequelize';
+import {
+  presentGroup,
+  presentPolicies,
+  presentUser,
+  presentGroupMembership,
+} from '../presenters';
+import { User, Event, Group, GroupUser } from '../models';
+import policy from '../policies';
+
+const { authorize } = policy;
+const router = new Router();
+
+router.post('groups.list', auth(), pagination(), async ctx => {
+  const user = ctx.state.user;
+  let groups = await Group.findAll({
+    where: {
+      teamId: user.teamId,
+    },
+    order: [['updatedAt', 'DESC']],
+    offset: ctx.state.pagination.offset,
+    limit: ctx.state.pagination.limit,
+  });
+
+  ctx.body = {
+    pagination: ctx.state.pagination,
+    data: groups.map(presentGroup),
+    policies: presentPolicies(user, groups),
+  };
+});
+
+router.post('groups.info', auth(), async ctx => {
+  const { id } = ctx.body;
+  ctx.assertUuid(id, 'id is required');
+
+  const user = ctx.state.user;
+  const group = await Group.findByPk(id);
+  authorize(user, 'read', group);
+
+  ctx.body = {
+    data: presentGroup(group),
+    policies: presentPolicies(user, [group]),
+  };
+});
+
+router.post('groups.create', auth(), async ctx => {
+  const { name } = ctx.body;
+  ctx.assertPresent(name, 'name is required');
+
+  const user = ctx.state.user;
+
+  authorize(user, 'create', Group);
+  const group = await Group.create({
+    name,
```

Comment:
Thoughts on making name unique within the team?

Reply:
yep, good idea